internal/pkg/services/graphql-service.go

```diff
@@ -0,0 +1,45 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package services
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+const (
+	serviceName = "graphql-server"
+	binaryName  = "cartesi-rollups-graphql-server"
+)
+
+type GraphQLService struct{}
+
+func (g GraphQLService) Start(ctx context.Context) error {
+	cmd := exec.Command(binaryName)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	go func() {
+		<-ctx.Done()
+		fmt.Printf("%v: %v\n", g.String(), ctx.Err())
+		cmd.Process.Signal(syscall.SIGTERM)
+	}()
+
+	err := cmd.Wait()
+	if err != nil && cmd.ProcessState.ExitCode() != int(syscall.SIGTERM) {
```

Comment:
Shouldn't this && be and assert? What if the ExitCode is not different but err is not nil?

Reply:
I don't like asserting here because this is an external process. I'd rather return the error and leave the decision to the callee (probably just log the error).

Reply:
There's the edge case where the process ends because of a `SIGTERM` that we didn't trigger. But considering that this is a temporary setup and will be thrown away after we fully rebuild the node in Go, it shouldn't be a significant concern.